### PR TITLE
 Large template rendering fix by removal of buffer pooling

### PR DESCRIPTION
**Pull request by wik8wz**

### CODE REVIEW

**Comment by toddams** [comment]
Thank you for your PR, looks great.
Please, solve these minor PR discussions and we'll merge it

**Comment by wik8wz** [comment]
@toddams 
> Please, solve these minor PR discussions and we'll merge it

Thank you for kind words. Do we have any discussions? Looks like you are the only one commenting

File: src/RazorLight/Internal/Buffering/NoPoolingViewBufferScope.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        /// <inheritdoc />
+        public ViewBufferValue[] GetPage(int pageSize)
+        {
+            if (pageSize <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(pageSize));
+            }
+
+            ViewBufferValue [] segment = new ViewBufferValue[Math.Max(pageSize, MinimumSize)];
+            _allBuffers.Add(segment);
```
Comment (jzabroski) [inline src/RazorLight/Internal/Buffering/NoPoolingViewBufferScope.cs]: I think this is a memory leak.

**Comment by jzabroski** [comment]
I'm going to close this PR, and re-write it to use safe pooling.

**Pull request closed without merging**